src/collection.js

```diff
@@ -157,10 +165,10 @@ export default class Collection {
       return Promise.reject(new Error("Record is not an object."));
     }
     const newRecord = Object.assign({}, record, {
-      id:      options.synced || options.forceUUID ? record.id : uuid4(),
+      id:      options.synced || options.forceUUID ? record.id : this._idSchema.generate(),
```

Comment:
I am not sure I understand this condition. ``if forceUUID then record.id`` I though forceUUID was to force an UUID generation.